warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/reindex/ShardReindexMapper.java

```diff
@@ -0,0 +1,764 @@
+            }
+
+            this.generateMetadata = config.getBoolean(GENERATE_METADATA, this.generateMetadata);
+
+            try {
+                this.event = new RawRecordContainerImpl();
+                if (eventOverride != null) {
+                    log.info("creating event override: " + this.eventOverride);
+                    this.event = (RawRecordContainer) Class.forName(this.eventOverride).getDeclaredConstructor().newInstance();
+                    if (this.event instanceof Configurable) {
+                        ((Configurable) this.event).setConf(config);
+                    }
+                }
+            } catch (ClassNotFoundException | InvocationTargetException | InstantiationException | IllegalAccessException | NoSuchMethodException e) {
+                throw new RuntimeException("Could not create event of type: " + this.eventOverride, e);
+            }
+
+            this.enableReindexCounters = config.getBoolean(ENABLE_REINDEX_COUNTERS, this.enableReindexCounters);
+            this.dumpCounters = config.getBoolean(DUMP_COUNTERS, this.dumpCounters);
+            if (this.enableReindexCounters) {
+                this.counters = new HashMap<>();
+            }
+
+            this.batchMode = BatchMode.valueOf(config.get(BATCH_MODE, this.batchMode.toString()));
+            if (this.batchMode != BatchMode.NONE) {
+                batchValues = new HashMap<>();
+            }
+        }
+    }
+
+    @Override
+    protected void cleanup(Context context) throws IOException, InterruptedException {
+        // process any remaining batch data
+        if (this.batchMode != BatchMode.NONE && this.batchValues.size() > 0) {
+            try {
+                processBatch(context);
+            } catch (IOException | InterruptedException e) {
+                throw new RuntimeException("Could not process final batch", e);
+            }
+        }
+
+        if (this.generateMetadata && this.indexHandler.getMetadata() != null) {
+            for (BulkIngestKey bik : this.indexHandler.getMetadata().getBulkMetadata().keySet()) {
+                for (Value v : this.indexHandler.getMetadata().getBulkMetadata().get(bik)) {
+                    context.write(bik, v);
+                }
+            }
+
+        }
+
+        // output counters if used
+        if (this.enableReindexCounters) {
+            for (String counterGroup : this.counters.keySet()) {
+                Map<String,Long> groupCounters = this.counters.get(counterGroup);
+                for (String counter : groupCounters.keySet()) {
+                    Long value = groupCounters.get(counter);
+                    // optionally dump the counters to logs
+                    if (this.dumpCounters) {
+                        log.info("COUNTER " + counterGroup + " " + counter + " " + value);
+                    } else {
+                        context.getCounter(counterGroup, counter).increment(value);
```

Comment:
The comment suggests that the dump counter logs in addition to incrementing but here we log OR increment. I like always incrementing.

Reply:
Updated the comment. This feature was required to support cases where the counters would overrun the MR limits and cause jobs to fail.

Reply:
Got it. Thanks!